egs2/TEMPLATE/spk1/spk.sh

```diff
@@ -391,3 +399,43 @@ if [ ${stage} -le 4 ] && [ ${stop_stage} -ge 4 ]; then
             --output_dir "${spk_exp}" \
             ${_opts} ${spk_args}
 fi
+
+if [ ${stage} -le 5 ] && [ ${stop_stage} -ge 5 ]; then
+    echo "Stage 5: Inference."
+
+    infer_exp=${spk_exp}/inference
+    _inference_dir=${data_feats}/${test_sets}
+    log "Spk inference started... log: '${infer_exp}/inference.log'"
+    if echo "${cuda_cmd}" | grep -e queue.pl -e queue-freegpu.pl &> /dev/null; then
+        # SGE can't include "/" in a job name
+        jobname="$(basename ${infer_exp})"
+    else
+        jobname="${infer_exp}/inference.log"
+    fi
+
+    _opts=
+    if [ -n "${spk_config}"  ]; then
+        # To generate the config file: e.g.
+        #   % python3 -m espnet2.bin.spk_train --print_config --optim adam
+        _opts+="--spk_train_config ${spk_config} "
+    fi
+
+    ${python} -m espnet2.bin.launch \
+        --cmd "${cuda_cmd} --name ${jobname}" \
+        --log ${infer_exp}/inference.log \
+        --ngpu ${ngpu} \
+        --num_nodes ${num_nodes} \
+        --init_file_prefix ${spk_exp}/.dist_init_ \
+        --multiprocessing_distributed true -- \
+        ${python} -m espnet2.bin.spk_inference \
```

Comment:
It is better to support both GPU and CPU inference with multiple processes (by specifying `--inference_nj`) like in other tasks.